### ISSUE TITLE
PageIterator seems to ignore request-options "WithAppOnly" on subsequent requests

### ISSUE THREAD

**Issue by Jan1503** [microsoftgraph/msgraph-sdk-dotnet#2036]
I'm in the process of migrating some code from v4 to v5 and encounter the same bug while using the NextPageRequest (v4) in the PageIterator (v5).

I have to get the Access-Packages from our Azure AD using graph and this has to be done with "AppOnly" application permissions.
For all other tasks, delegated permissions are needed so I don't have the Graph-Client injected with AppOnly in the first place.

So I just add the option "WithAppOnly()" to the request and all works well for the first request.

But as soon as the single-page limit is reached, in this case 100, the iterator kicks in and ignore the options from the request so the next call crashes with an unauthorized error because the request is done with the user-credentials instead.

Here's the entire function with some explanations to better understand the issue:
![image](https://github.com/microsoftgraph/msgraph-sdk-dotnet/assets/10504259/656813d9-c1ec-4b2f-99a1-da4ceab83978)

The behaviour is the same in the old v4 but there was an easy workaround to add the "WithAppOnly" to the "NextPageRequest".

Please show me how add the options to any subsequent request...I tried many workarounds but haven't found a working one yet.

**Comment by andrueastman** [comment]
Thanks for raising this @Jan1503,

Just to confirm, is `WithAppOnly()` a custom method you have to update the Option collection?

Any chance it works out if you have the following request configurator?

```cs
 subsequentRequest =>
                    {
                        subsequentRequest.RequestOptions.WithAppOnly(); // How to add ...WithAppOnly() to following request?
                        return subsequentRequest;
                    }
```

Alternatively, you can also call

```cs
subsequentRequest =>
                    {
                        subsequentRequest.AddRequestOptions( ... );// pass a list of IRequestOptions
                        return subsequentRequest;
                    }
```

**Comment by Jan1503** [comment]
No, `WithAppOnly()` is an official extension method in `Microsoft.Identity.Web.RequestOptionsExtension`.

It instructs the graph to use application-permissions for the next request instead of delegated-permissions.

As I said, the first request without the iterator works as expected but any subsequent requests fail, because the iterator ignores the request options configured in the first request.

This is also the case when using the "old" sdk as I have mentioned here without any reply:
[https://github.com/AzureAD/microsoft-identity-web/issues/2272](https://github.com/AzureAD/microsoft-identity-web/issues/2272)

**Comment by Jan1503** [comment]
@andrueastman 
Hey, any news on this or a workaround?
This stops me from updating my app!

**Comment by h3rmanj** [comment]
Ran into this as well.

Solved it by adding

```csharp
config =>
{
    var appOptions = new List<IRequestOption>();
    appOptions.WithAppOnly();
    config.AddRequestOptions(appOptions);

    return config;
}
```
